File: SYP.Web/Modules/ServerTypes/Email/EmailQueueColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { EmailPriority } from "../_Ext/EmailPriority";
import { EmailQueueRow } from "./EmailQueueRow";

export interface EmailQueueColumns {
    Id: Column<EmailQueueRow>;
    SmtpSettingsName: Column<EmailQueueRow>;
    Subject: Column<EmailQueueRow>;
    ToAddresses: Column<EmailQueueRow>;
    Priority: Column<EmailQueueRow>;
    Status: Column<EmailQueueRow>;
    RetryCount: Column<EmailQueueRow>;
    ScheduledAt: Column<EmailQueueRow>;
    SentAt: Column<EmailQueueRow>;
    InsertDate: Column<EmailQueueRow>;
}

export class EmailQueueColumns extends ColumnsBase<EmailQueueRow> {
    static readonly columnsKey = 'Email.EmailQueue';
    static readonly Fields = fieldsProxy<EmailQueueColumns>();
}

[EmailPriority]; // referenced types

File: SYP.Web/Modules/Email/EmailQueue/EmailQueueColumns.cs
namespace SYP.Email.Columns;

[ColumnsScript("Email.EmailQueue")]
[BasedOnRow(typeof(EmailQueueRow), CheckNames = true)]
public class EmailQueueColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight, Width(80)]
    public long Id { get; set; }

    [Width(100)]
    public string SmtpSettingsName { get; set; }

    [EditLink, Width(250)]
    public string Subject { get; set; }

    [Width(200)]
    public string ToAddresses { get; set; }

    [Width(100)]
    public int Priority { get; set; }

    [Width(120)]
    public int Status { get; set; }

    [Width(80)]
    public int RetryCount { get; set; }

    [Width(150)]
    public DateTime ScheduledAt { get; set; }

    [Width(150)]
    public DateTime SentAt { get; set; }

    [Width(150)]
    public DateTime InsertDate { get; set; }
}
